sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -129,34 +129,63 @@ object SetOperationPushDown extends Rule[LogicalPlan] {
     result.asInstanceOf[A]
   }
 
+  /**
+   * Splits the condition expression into small conditions by `And`, and partition them by
+   * deterministic, and finally recombine them by `And`.
+   */
```

Comment:
Let's add a comment to explain the meanings of returned Expressions (i.e. the first `Expression` is for deterministic expressions and the second `Expression` is for non-deterministic expressions).